backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java

```diff
@@ -1907,4 +1911,50 @@ public ProjectData searchByTag(String tag, User user) {
     public ProjectData searchByType(String type, User user) {
         return repository.searchByType(type, user);
     }
+
+    public ByteBuffer getReportDataStream(List<Project> documents, User user, boolean extendedByReleases)
+			throws TException {
+		try {
+			ThriftClients thriftClients = new ThriftClients();
+			ProjectExporter exporter = getProjectExporterObject(documents, user, extendedByReleases, thriftClients);
+			InputStream stream = exporter.makeExcelExport(documents);
+			return ByteBuffer.wrap(IOUtils.toByteArray(stream));
+		} catch (IOException e) {
+			throw new SW360Exception(e.getMessage());
+		}
+	}
+
+    private ProjectExporter getProjectExporterObject(List<Project> documents, User user, boolean extendedByReleases,
+            ThriftClients thriftClients) throws SW360Exception {
+    	return new ProjectExporter(thriftClients.makeComponentClient(),
+                thriftClients.makeProjectClient(), user, documents, extendedByReleases);
+    }
+
+    public String getReportInEmail(List<Project> documents, User user,
+			boolean extendedByReleases) throws TException {
+    	try {
+    		ThriftClients thriftClients = new ThriftClients();
+    		ProjectExporter exporter = getProjectExporterObject(documents, user, extendedByReleases, thriftClients);
+    		return exporter.makeExcelExportForProject(documents, user);
+    	}catch (IOException e) {
+			throw new SW360Exception(e.getMessage());
+		}
+		
+	}
+
+	public ByteBuffer downloadExcel(User user, boolean extendedByReleases, String token) throws SW360Exception {
```

Comment:
Please check the Indentation in this file

Reply:
Resolved